src/services/resource.js

```diff
@@ -626,6 +630,119 @@ module.exports = class resourceHelper {
+	 */
+	static async getDetails(resourceId) {
+		try {
+			let result = {
+				organization: {},
+			}
+
+			const resource = await resourceQueries.findOne({
+				id: resourceId,
+			})
+
+			if (!resource?.id) {
+				return responses.failureResponse({
+					message: 'RESOURCE_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			if (resource.blob_path) {
+				const response = await filesService.fetchJsonFromCloud(resource.blob_path)
+				if (
+					response.statusCode === httpStatusCode.ok &&
+					response.result &&
+					Object.keys(response.result).length > 0
+				) {
+					//modify the response as label value pair
+					let resultData = response.result
+
+					//get all entity types with entities
+					let entityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+						{
+							model: resource.type,
+							status: common.STATUS_ACTIVE,
+						},
+						resource.organization_id,
+						['id', 'value', 'label', 'has_entities']
+					)
+
+					if (entityTypes.length > 0) {
+						//create label value pair map
+						const entityTypeMap = entityTypes.reduce((map, type) => {
+							if (type.has_entities && Array.isArray(type.entities) && type.entities.length > 0) {
+								map[type.value] = type.entities
+									.filter((entity) => entity.status === common.STATUS_ACTIVE)
+									.map((entity) => ({ label: entity.label, value: entity.value.toLowerCase() }))
+							}
+							return map
+						}, {})
+
+						for (let entityType of entityTypes) {
+							const key = entityType.value
+							// Skip the entity type if entities are not available
+							if (
+								entityType.has_entities &&
+								entityType.entities &&
+								entityType.entities.length > 0 &&
+								resultData.hasOwnProperty(key)
+							) {
+								const value = resultData[key]
+								// If the value is already in label-value pair format, skip processing
+								if (utils.isLabelValuePair(value) || value === '') {
+									continue
+								}
+
+								// get the entities
+								const validEntities = entityTypeMap[key] || []
+
+								if (Array.isArray(value)) {
+									// Map each item in the array to a label-value pair, if it exists in validEntities
+									resultData[key] = value.map((item) => {
+										const match = validEntities.find(
+											(entity) => entity.value === item.toLowerCase()
+										)
+										return match || { label: item, value: item.toLowerCase() }
+									})
+								} else {
+									// If the value is a single item, find it in validEntities
+									const match = validEntities.find((entity) => entity.value === value.toLowerCase())
+									resultData[key] = match || { label: value, value: value.toLowerCase() }
+								}
+							}
+						}
+
+						result = { ...result, ...resultData }
+					}
+				}
+			}
+
+			//get organization details
+			let organizationDetails = await userRequests.fetchDefaultOrgDetails(resource.organization_id)
```

Comment:
@priyanka-TL why do we have function called "fetchDefaultOrgDetails" it should be just "fetchOrg". Also it has to be via org extension first.